test/integration/puppetlabs/services/certificate_authority/certificate_authority_int_test.clj

```diff
@@ -1082,15 +1103,32 @@
            {:allow-auto-renewal true}
            :authorization
            {:allow-header-cert-info true}}
-          (let [header-cert (ring-codec/url-encode (slurp (str bootstrap/server-conf-dir "/ssl/certs/localhost.pem")))
+          (let [generated-cert-info (generate-and-sign-a-cert! "foobar")
+                signed-cert-file (ks/temp-file)
+                _ (spit signed-cert-file (:signed-cert generated-cert-info))
+                header-cert (ring-codec/url-encode (:signed-cert generated-cert-info))
+                _ (Thread/sleep 1000)
                 response (http-client/post
                            "https://localhost:8140/puppet-ca/v1/certificate_renewal"
                            {:headers     {"x-client-cert" header-cert}
-                            :ssl-cert    (str bootstrap/server-conf-dir "/ssl/certs/localhost.pem")
-                            :ssl-key      (str bootstrap/server-conf-dir "/ssl/private_keys/localhost.pem")
                             :ssl-ca-cert (str bootstrap/server-conf-dir "/ca/ca_crt.pem")
                             :as          :text})]
-            (is (= 501 (:status response))))))
+            (is (= 200 (:status response)))
+            (let [renewed-cert-pem (:body response)
+                  renewed-cert-file (ks/temp-file)
+                  _ (spit renewed-cert-file renewed-cert-pem)
+                  renewed-cert (ssl-utils/pem->cert renewed-cert-file)
+                  signed-cert (ssl-utils/pem->cert signed-cert-file)]
+              (testing "serial number has been incremented"
+                (is (< (.getSerialNumber signed-cert) (.getSerialNumber renewed-cert))))
+              (testing "not before time stamps have changed"
+                (is (= -1 (.compareTo (.getNotBefore signed-cert) (.getNotBefore renewed-cert)))))
```

Comment:
It looks like Date has some more intuitive methods for comparison than compareTo. While I get what this is doing, I think equal, after, or before would be more readable.